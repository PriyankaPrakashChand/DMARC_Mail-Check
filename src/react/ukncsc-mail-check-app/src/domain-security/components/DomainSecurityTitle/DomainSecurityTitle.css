.DomainSecurityTitle--header {
  word-break: break-word;
  word-wrap: break-word;
  display: inline-block;
}

.DomainSecurityTitle--icon {
  display: inline-block;
  margin-left: 5px;
}
