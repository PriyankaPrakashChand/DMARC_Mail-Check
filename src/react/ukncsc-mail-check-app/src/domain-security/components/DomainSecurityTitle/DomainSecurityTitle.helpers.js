// eslint-disable-next-line
export const getTitleIconProps = (failures, warnings, inconclusives) => {
  if (failures && !!failures.length) {
    return { name: 'exclamation circle', color: 'red' };
  }

  if (warnings && !!warnings.length) {
    return { name: 'exclamation triangle', color: 'yellow' };
  }

  if (inconclusives && !!inconclusives.length) {
    return { name: 'question circle', color: 'grey' };
  }

  return null;
};
