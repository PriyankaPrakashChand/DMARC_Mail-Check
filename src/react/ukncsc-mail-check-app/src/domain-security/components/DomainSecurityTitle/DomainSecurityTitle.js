import React from 'react';
import PropTypes from 'prop-types';
import { Header, Icon, Loader } from 'semantic-ui-react';
import { getTitleIconProps } from './DomainSecurityTitle.helpers';

import './DomainSecurityTitle.css';

const DomainSecurityTitle = ({
  as,
  title,
  loading,
  subtitle,
  failures,
  warnings,
  inconclusives,
  children,
}) => {
  const icon = getTitleIconProps(failures, warnings, inconclusives);

  return (
    <React.Fragment>
      <Header as={as} className="DomainSecurityTitle--header">
        {title}
      </Header>
      <div className="DomainSecurityTitle--icon">
        {loading && <Loader active inline />}
        {icon && (
          <Icon
            {...icon}
            size="big"
            style={{ lineHeight: as === 'h1' ? '0.3' : '0.4' }}
          />
        )}
      </div>
      {subtitle && <Header as="h2">{subtitle}</Header>}
      {children}
    </React.Fragment>
  );
};

DomainSecurityTitle.defaultProps = {
  as: 'h1',
  loading: false,
  subtitle: null,
  failures: [],
  warnings: [],
  inconclusives: [],
  pending: false,
};

DomainSecurityTitle.propTypes = {
  as: PropTypes.oneOf(['h1', 'h2', 'h3']),
  title: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  subtitle: PropTypes.string,
  failures: PropTypes.arrayOf(PropTypes.string),
  warnings: PropTypes.arrayOf(PropTypes.string),
  inconclusives: PropTypes.arrayOf(PropTypes.string),
  pending: PropTypes.bool,
};

export default DomainSecurityTitle;
