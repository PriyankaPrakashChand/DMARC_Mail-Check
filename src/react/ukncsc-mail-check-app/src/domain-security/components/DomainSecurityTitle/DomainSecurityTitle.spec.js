/* eslint-disable import/first */
import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';
import DomainSecurityTitle from './DomainSecurityTitle';

describe('DomainSecurityTitle', () => {
  let container;

  describe('when failures are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" failures={['err!']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a red exclamation circle', () =>
      expect(
        container.getElementsByClassName('red exclamation circle icon')
      ).toHaveLength(1));
  });

  describe('when children are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a">
          <p>Last checked one day ago</p>
        </DomainSecurityTitle>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should display render the children', () =>
      expect(container.getElementsByTagName('p')[0]).toHaveTextContent(
        'Last checked one day ago'
      ));
  });

  describe('when warnings are present without failures', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" warnings={['warn!']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a yellow exclamation triangle', () =>
      expect(
        container.getElementsByClassName('yellow exclamation triangle icon')
      ).toHaveLength(1));
  });

  describe('when inconclusives are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" inconclusives={['inconclusive!']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a grey question circle', () =>
      expect(
        container.getElementsByClassName('grey question circle icon')
      ).toHaveLength(1));
  });

  describe('when no warnings, failures or inconclusives are present', () => {
    beforeEach(() => {
      ({ container } = render(<DomainSecurityTitle title="a" />));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not have any icons', () =>
      expect(container.getElementsByTagName('i')).toHaveLength(0));
  });
});
