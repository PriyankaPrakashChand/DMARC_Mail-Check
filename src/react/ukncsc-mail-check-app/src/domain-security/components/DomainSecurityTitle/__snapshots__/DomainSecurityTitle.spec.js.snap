// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityTitle when children are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header DomainSecurityTitle--header"
  >
    a
  </h1>
  <div
    class="DomainSecurityTitle--icon"
  />
  <p>
    Last checked one day ago
  </p>
</div>
`;

exports[`DomainSecurityTitle when failures are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header DomainSecurityTitle--header"
  >
    a
  </h1>
  <div
    class="DomainSecurityTitle--icon"
  >
    <i
      style="line-height: 0.3;"
      aria-hidden="true"
      class="red exclamation circle big icon"
    />
  </div>
</div>
`;

exports[`DomainSecurityTitle when inconclusives are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header DomainSecurityTitle--header"
  >
    a
  </h1>
  <div
    class="DomainSecurityTitle--icon"
  >
    <i
      style="line-height: 0.3;"
      aria-hidden="true"
      class="grey question circle big icon"
    />
  </div>
</div>
`;

exports[`DomainSecurityTitle when no warnings, failures or inconclusives are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header DomainSecurityTitle--header"
  >
    a
  </h1>
  <div
    class="DomainSecurityTitle--icon"
  />
</div>
`;

exports[`DomainSecurityTitle when warnings are present without failures it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header DomainSecurityTitle--header"
  >
    a
  </h1>
  <div
    class="DomainSecurityTitle--icon"
  >
    <i
      style="line-height: 0.3;"
      aria-hidden="true"
      class="yellow exclamation triangle big icon"
    />
  </div>
</div>
`;
